packages/edit-post/src/components/sidebar/settings-sidebar/index.js

```diff
@@ -62,43 +67,80 @@ const SettingsSidebar = () => {
 			).getShortcutRepresentation( 'core/edit-post/toggle-sidebar' );
 			return {
 				sidebarName: sidebar,
+				isSidebarOpen: isOpen,
 				keyboardShortcut: shortcut,
 				isTemplateMode: select( editPostStore ).isEditingTemplate(),
 			};
+		}, [] );
+
+	const Content = () => {
+		// Because `PluginSidebarEditPost` renders a `ComplementaryArea`, we
+		// need to forward the `Tabs` context so it can be passed through the
+		// underlying slot/fill.
+		const tabsContextValue = useContext( Tabs.Context );
+
+		return (
+			<PluginSidebarEditPost
+				identifier={ sidebarName }
+				header={
+					<Tabs.Context.Provider value={ tabsContextValue }>
+						<SettingsHeader sidebarName={ sidebarName } />
+					</Tabs.Context.Provider>
+				}
+				closeLabel={ __( 'Close Settings' ) }
+				headerClassName="edit-post-sidebar__panel-tabs"
+				/* translators: button label text should, if possible, be under 16 characters. */
+				title={ __( 'Settings' ) }
+				toggleShortcut={ keyboardShortcut }
+				icon={ isRTL() ? drawerLeft : drawerRight }
+				isActiveByDefault={ SIDEBAR_ACTIVE_BY_DEFAULT }
+			>
+				<Tabs.Context.Provider value={ tabsContextValue }>
+					<Tabs.TabPanel
+						id={ 'edit-post/document' }
+						focusable={ false }
+					>
+						{ ! isTemplateMode && (
+							<>
+								<PostStatus />
+								<PluginDocumentSettingPanel.Slot />
+								<LastRevision />
+								<PostTaxonomies />
+								<FeaturedImage />
+								<PostExcerpt />
+								<DiscussionPanel />
+								<PageAttributes />
+								<MetaBoxes location="side" />
+							</>
+						) }
+						{ isTemplateMode && <TemplateSummary /> }
+					</Tabs.TabPanel>
+					<Tabs.TabPanel id={ 'edit-post/block' } focusable={ false }>
+						<BlockInspector />
+					</Tabs.TabPanel>
+				</Tabs.Context.Provider>
+			</PluginSidebarEditPost>
+		);
+	};
+
+	const { openGeneralSidebar } = useDispatch( editPostStore );
+
+	const onTabSelect = useCallback(
+		( newSelectedTabId ) => {
+			if ( !! newSelectedTabId ) {
+				openGeneralSidebar( newSelectedTabId );
+			}
 		},
-		[]
+		[ openGeneralSidebar ]
 	);
 
 	return (
-		<PluginSidebarEditPost
-			identifier={ sidebarName }
-			header={ <SettingsHeader sidebarName={ sidebarName } /> }
-			closeLabel={ __( 'Close Settings' ) }
-			headerClassName="edit-post-sidebar__panel-tabs"
-			/* translators: button label text should, if possible, be under 16 characters. */
-			title={ __( 'Settings' ) }
-			toggleShortcut={ keyboardShortcut }
-			icon={ isRTL() ? drawerLeft : drawerRight }
-			isActiveByDefault={ SIDEBAR_ACTIVE_BY_DEFAULT }
+		<Tabs
+			selectedTabId={ isSidebarOpen ? sidebarName : null }
```

Comment:
We should probably leave a comment to why we're passing `null` when the sidebar isn't open, to spare the same headache to the next developer touching this file

Reply:
done!